libs/sdk-ui-ext/src/internal/components/dialogs/embedInsightDialog/EmbedInsightDialog.tsx

```diff
@@ -0,0 +1,162 @@
+// (C) 2022 GoodData Corporation
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { IColorPalette, IInsight } from "@gooddata/sdk-model";
+import {
+    CodeOptionType,
+    CodeLanguageType,
+    EmbedInsightDialogBase,
+    InsightCodeType,
+    IOptionsByDefinition,
+    Overlay,
+    getDefaultOptions,
+    getHeightWithUnits,
+    IOptionsByReference,
+} from "@gooddata/sdk-ui-kit";
+import { IAnalyticalBackend, IUserWorkspaceSettings } from "@gooddata/sdk-backend-spi";
+import { FullVisualizationCatalog } from "../../VisualizationCatalog";
+import { insightViewCodeGenerator } from "../../../utils/embeddingInsightViewCodeGenerator/insightViewCodeGenerator";
+
+const InsightViewPropertiesLink =
+    "https://sdk.gooddata.com/gooddata-ui/docs/visualization_component.html#properties";
+
+/**
+ * @internal
+ */
+export interface IEmbedInsightDialogProps {
+    codeType: InsightCodeType;
+    insight: IInsight;
+    locale?: string;
+    backend?: IAnalyticalBackend;
+    settings?: IUserWorkspaceSettings;
+    colorPalette?: IColorPalette;
+
+    onClose: () => void;
+    onCopyCode: () => void;
```

Comment:
Some idea: would it be useful, if this callback also includes the code as a parameter?

Reply:
Done

Reply:
You say done, but the callback still does not accept the code. Did you forget to push some changes maybe?

Reply:
Now done